### ISSUE TITLE
Add support for explicitly choosing an aggregator in Query Builder

### ISSUE THREAD

**Issue by grantfitzsimmons** [specify/specify7#4620]
The feature request "Add support for explicitly choosing a formatter in Query Builder" (https://github.com/specify/specify7/issues/1959) did not cover choosing an aggregator.

@maxpatiiuk said in [this comment](https://github.com/specify/specify7/issues/1959#issuecomment-1579769627) that at one point it was
>Fixed both for formatters and aggregators

[This code](https://github.com/specify/specify7/commit/231ac56ca81e24debfb3d54b15426bdde39edc45#diff-beffe9e3545186073f701dcc4e18ee942db4e7f7a0439fec2c3fdb8d5f281e26R376-R396) conditionally renders a `QueryFieldFormatter` component based on the `fieldMeta` properties. If `fieldType` is either 'formatter' or 'aggregator' and `tableName` is a string, it displays the `QueryFieldFormatter` component with specific props like `formatter`, `tableName`, `type`, and `onChange`.

Theoretically it already _should_ support this for aggregators, but either due to a bug in the existing code or an oversight, this is not functional at the moment.

This indicates to me that the intention was to have this existing feature request cover aggregators as well (as the logic should be nearly identical).

**Expected Behavior**
* When adding an `(aggregated)` line to a query, the user should be able to explicitly define which aggregation to use
* The gear icon should only appear to the right of the `(aggregated)` line if there are two or more aggregators configured for that table
* This should behave **identically** to how the user defines an explicit formatter (added back in https://github.com/specify/specify7/pull/4615)

This is arguably a bug in the existing implementation, but because it is not functional and outside the scope of the previous issue, I have opened this one. I do see this as one and the same as #1959 at its core.

**Comment by CarolineDenis** [comment]
https://github.com/specify/specify7/blob/3fcba9ba397c18faa9b3c7e68c4493942ea3909d/specifyweb/frontend/js_src/lib/components/QueryBuilder/Line.tsx#L140

```ts
const dataModelField = isFormatted
        ? undefined
        : getTable(tableName ?? '')?.getField(fieldName);
```
   
https://github.com/specify/specify7/blob/3fcba9ba397c18faa9b3c7e68c4493942ea3909d/specifyweb/frontend/js_src/lib/components/QueryBuilder/Line.tsx#L168
     
```ts
else if (isMapped)
        fieldType =
          dataModelField?.isRelationship && relationshipIsToMany(dataModelField)
            ? 'aggregator'
            : 'formatter';
```
            
In the case where we add a Formatted resource or Aggregated resource, `isFormatted` will be true. (isFormatted checks if the field starts with `'-'`, in this case we have` '-formatted'` and `'-aggregated'`) 

Therefore, `dataModelField` will be undefined in both cases. ==> `isFormatted === true ? undefined `

Since dataModelField is always undefined, `dataModelField?.isRelationship` or `relationshipIsToMany(dataModelField)` will also always be undefined, 

In conclusion the value of fieldType will always be 'formatted' with this code

**Comment by CarolineDenis** [comment]
Fixed in https://github.com/specify/specify7/pull/4615/commits/ddebb7a3daf98c30713e70be469019e5dd46d9f1

**Comment by acwhite211** [comment]
This is fixed now on the back-end in #4646
https://github.com/specify/specify7/pull/4646#issuecomment-2000269004
> The problem seems to be that the back-end is ignoring the formatter name in the request. It is just getting the aggragator and then getting the formatter that the aggregator is pointing to. I changed the code so that it takes into account the 'formatter name' that is passed in the query builder request.
